### PR TITLE
Enhancements in JSON Field Population

### LINKED ISSUE #21543: Ensure JSON for Vulnerability Inventory Indexing Contains No Empty Fields
#### Description

This issue focuses on improving the quality of the JSON data used to index the vulnerability inventory into Wazuh indexer. Currently, there are instances where fields within this JSON structure may contain empty values. To enhance data consistency and integrity, it is essential to ensure that the JSON data being indexed contains no empty fields.

#### User Stories

- As a data consumer, I want to receive well-structured JSON data with no empty fields to avoid issues with data processing.
- As a developer, I want to enhance the quality of data indexed into Wazuh Indexer by eliminating empty fields.

#### Tasks

1. **Identify Empty Fields:**
   - Analyze the existing code responsible for constructing the JSON used for indexing the vulnerability inventory.
   - Identify cases where fields may have empty or null values.

2. **Field Validation:**
   - Implement validation checks to ensure that fields within the JSON structure do not contain empty values.
   - Handle scenarios where empty fields are encountered, either by providing default values or excluding them from the JSON.

#### Definition of Done

- The code responsible for constructing the JSON data for vulnerability inventory indexing has been updated to prevent empty fields.
- Validation checks have been implemented and tested to ensure no empty fields are present in the JSON.

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|Closes #21543 |

## Objective
This pull request introduces significant improvements to how fields are populated in JSON objects within our codebase. The main tasks include removing all assignments in scenarios where a field cannot be populated and creating a versatile helper function, `populateField`, to streamline the process of adding data to JSON objects. This refactor aims to enhance code maintainability, readability, and efficiency in handling JSON data operations. Additionally, unit tests (UT) have been refactored to align with these changes, ensuring the robustness and reliability of the new implementation.

## Description
- **Removing Unnecessary Assignments**: This change involves modifying the logic to prevent assigning values to JSON fields when it is determined that these fields cannot be populated due to various constraints (e.g., missing data, errors). This approach reduces clutter and potential errors in JSON handling.
- **Helper Function Implementation**: The `populateField` function template has been designed to offer a generic, flexible solution for populating JSON objects with various data types. This utility simplifies the code and reduces duplication by providing a single, unified way to add data to JSON fields.
  
  Signature: 
  ```cpp
  template<typename T>
  void populateField(nlohmann::json& json, const nlohmann::json::json_pointer& key, T&& value);
  ```
- **Refactoring Unit Tests**: Due to the introduction of `populateField` and the change in how JSON fields are populated, existing unit tests have been refactored. These modifications ensure that tests accurately reflect the new logic and continue to verify the correctness and stability of JSON handling.

## Quality Assurance
- Comprehensive testing has been conducted to validate the functionality and reliability of the `populateField` helper function across various scenarios and data types.
- The refactored unit tests cover a broad range of cases to ensure that the new approach to JSON field population does not introduce regressions and adheres to expected behaviors.

## Implementation Details
- The logic for conditional field assignments in JSON objects has been revised to eliminate unnecessary operations.
- The `populateField` function has been implemented as a template to accommodate different types of values being assigned to JSON fields, enhancing code reusability and efficiency.
- Unit tests have been extensively refactored to align with the new JSON field population strategy and validate the changes thoroughly.

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp [hunk 1]
```diff
@@ -29,6 +29,32 @@ class TDetailsAugmentation final : public AbstractHandler<std::shared_ptr<TScanC
...
+     * This function checks if the provided value is a non-empty string and populates the specified JSON field with it.
+     * If the value is an empty string, it returns without making any changes to the JSON object.
+     *
+     * @tparam T The type of the value to populate the JSON field with.
+     * @param json The JSON object to populate.
+     * @param key The JSON pointer specifying the field.
+     * @param value The value to populate the field with (as std::string_view).
+     */
+    template<typename T>
+    void populateField(nlohmann::json& json, const nlohmann::json::json_pointer& key, const std::string_view value)
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp:46]: not all values should be a string, the json values could be numbers, floats, null, and... strings.

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp [hunk 2]
```diff
@@ -29,6 +29,32 @@ class TDetailsAugmentation final : public AbstractHandler<std::shared_ptr<TScanC
...
+    {
+        if constexpr (std::is_same_v<T, std::string>)
+        {
+            if (value.empty())
+            {
+                return; // Do nothing for empty strings
+            }
+        }
+
+        json[key] = std::string(value); // Convert std::string_view to std::string if needed
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp:56]: same as above

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp (hunk 1 above)
Comment (Dwordcito) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp:46]: probably we can use:

```cpp
template<typename T>
void populateField(nlohmann::json& json, const nlohmann::json::json_pointer& key, T&& value)
{
    if constexpr (std::is_same_v<std::remove_cv_t<std::remove_reference_t<T>>, std::string_view> || std::is_same_v<T, std::string>)
    {
        if (!value.empty())
        {
            json[key] = value;
        }
    }
    else
    {
        json[key] = value;
    }
}
```

Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp]: Done!

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp (hunk 2 above)
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp]: Thx!

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**